core/trino-parser/src/main/java/io/trino/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -833,7 +833,7 @@ protected Expression visitRowDataType(RowDataType node, Context<C> context)
                     }
                 }
 
-                if (dataType != field.getType() || name != field.getName()) {
+                if (dataType != field.getType() || !name.equals(field.getName())) {
```

Comment:
I think we shouldn't modify this. Please note that `field.getName()` is a Node, and not just String. If we replace reference equality by `equals`, it will be possible to lose the Identifier's NodeLocation during rewrite, and still consider the tree unrewritten, which is not consistent with the spirit of ExpressionTreeRewriter.